teammapper-backend/src/map/controllers/maps.gateway.ts

```diff
@@ -167,6 +168,41 @@ export class MapsGateway implements OnGatewayDisconnect {
     return true
   }
 
+  @UseGuards(EditGuard)
+  @SubscribeMessage('undoRedoMapChanges')
+  async undoRedoMapChanges(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() request: IMmpClientUndoRedoRequest
+  ): Promise<boolean> {
+    if (!(await this.mapsService.findMap(request.mapId)))
+      return Promise.resolve(false)
+    if (!request.diff)
+      return Promise.resolve(false)
+
+    // Disconnect all clients temporarily
+    // Emit an event so clients can display a notification
+    this.server.to(request.mapId).emit('clientNotification', { clientId: client.id, message: 'TOASTS.WARNINGS.MAP_UPDATE_IN_PROGRESS', type: 'warning' })
```

Comment:
disconnecting all clients just because a change like a name is reverted feels a big drastic? For the case of importing a whole different map it makes sense, but just updating a name or coordinates of a node should not drop all connections of the clients.

Reply:
Clients no longer get disconnected when executing undo/redo, seems to work fine.